synthesizer/program/src/logic/instruction/operation/commit.rs

```diff
@@ -110,6 +111,50 @@ impl<N: Network, const VARIANT: u8> CommitInstruction<N, VARIANT> {
     }
 }
 
+// This code is nearly identical in `execute` and `evaluate`; we
+// extract it here in a macro.
+//
+// The `$q` parameter allows us to wrap a value in `Result::Ok`, since
+// the `Aleo` functions don't return a `Result` but the `Network` ones do.
+macro_rules! do_commit {
+    ($N: ident, $variant: expr, $destination_type: expr, $input: expr, $randomizer: expr, $ty: ty, $q: expr) => {{
+        let func = match $variant {
+            0 => $N::commit_to_group_bhp256,
+            1 => $N::commit_to_group_bhp512,
+            2 => $N::commit_to_group_bhp768,
+            3 => $N::commit_to_group_bhp1024,
+            4 => $N::commit_to_group_ped64,
+            5 => $N::commit_to_group_ped128,
+            6.. => bail!("Invalid 'commit' variant: {}", $variant),
+        };
+
+        let literal_output: $ty = $q(func(&$input.to_bits_le(), $randomizer))?.into();
+        literal_output.cast_lossy($destination_type)?
+    }};
+}
+
+/// Evaluate a commit operation.
+///
+/// This allows running the commit without the machinery of stacks and registers.
+/// This is necessary for the Leo interpeter.
+pub fn evaluate_commit<N: Network>(
+    variant: CommitVariant,
+    input: &Value<N>,
+    randomizer: &Scalar<N>,
+    destination_type: LiteralType,
+) -> Result<Literal<N>> {
+    evaluate_commit_private(variant as u8, input, randomizer, destination_type)
```

Comment:
Nit: in snarkVM we more often use the suffix `_internal`, because the word "private" is used for cryptography and alerts and fatigues when validating security issues.